### PR TITLE
Statmap inj efficiency

### PR DESCRIPTION
**Pull request by GarethCabournDavies**
the statmap inj job takes a big proportion of the time in the CI search jobs.

In this, I add an --ignore-inclusive flag, as we don't use inclusive IFAR for injections anymore anyway, so it skips these calculations and skips a lot of computation which is unnecessary.

Added this flag to the search CI to check that it still works

### CODE REVIEW

File: bin/all_sky_search/pycbc_coinc_statmap_inj [hunk 1]
```diff
@@ -118,25 +121,30 @@ if len(zdata) > 0:
...
 
     # most of the triggers are here so presort to speed up later sorting
     bsort = back_stat.argsort()
     dec_fac = dec_fac[bsort]
     back_stat = back_stat[bsort]
 
+    if args.ignore_inclusive:
+        # If we are not calculating inclusive IFAR, we can skip this last bit
+        logging.info("Done!")
+        exit()
```
Comment (spxiwh) [inline bin/all_sky_search/pycbc_coinc_statmap_inj:147]: I think using this is discouraged, so maybe just put the whole block in a `if not args.inclusive` or (better) move to a function and simply don't call it.

Line 181 and 182 also shouldn't be set if ignoring inclusive incase the presence of an [empty] array is taken to indicate that this is expected.

Comment (GarethCabournDavies) [reply bin/all_sky_search/pycbc_coinc_statmap_inj:147]: The empty array is a good idea, and think that should fix the CI failure

**Comment by GarethCabournDavies** [comment]
Note on the timings, running on one of the CIT head nodes:
 - Running without the flag took 52 minutes
 - Running with the flag took 8 seconds

File: bin/all_sky_search/pycbc_coinc_statmap_inj (hunk 1 above)
Comment (GarethCabournDavies) [reply bin/all_sky_search/pycbc_coinc_statmap_inj]: Is a for loop with an `if ...: continue` and an `else` okay? It removes the exit(), and prevents an (IMO) ugly big `if` block.

Comment (spxiwh) [reply bin/all_sky_search/pycbc_coinc_statmap_inj]: Yes, I missed that this was in a for loop, `continue` is better then

File: bin/all_sky_search/pycbc_coinc_statmap_inj [hunk 2]
```diff
@@ -164,6 +171,9 @@ if len(zdata) > 0:
         ifar[i] = background_time / (fnlouder[i] + 1)
         fap[i] = 1 - numpy.exp(- coinc_time / ifar[i])
         logging.info('processed %s, %s' % (i, fstat))
+    else:
```
Comment (spxiwh) [inline bin/all_sky_search/pycbc_coinc_statmap_inj:174]: The code will *always* end up in this else block though! (Perhaps avoid `for/ else` unless it's needed as the syntax here is confusing). I also thought that the idea was *not* to populate the `foreground/ifar` and `foreground/fap` columns if skipping exclusive, which is not what is being done here?

Comment (GarethCabournDavies) [reply bin/all_sky_search/pycbc_coinc_statmap_inj:174]: I think it won't unless the flag is set

I've put another if statement in to make it more obvious than a for/else

**Comment by GarethCabournDavies** [comment]
I will close this in favour of removing all inclusive parts of the injection FAR calculation throughout the workflow (as discussed on slack)

**Pull request closed without merging**